modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java

```diff
@@ -45,7 +46,7 @@
   private final ClassLoader classLoader;
   private final Map<String, Class<?>> extensionsByName = new HashMap<>();
   private final Map<String, ExtensionModel> resolvedModels = new HashMap<>();
-  private TypeCatalog typeCatalog;
+  private LazyValue<TypeCatalog> typeCatalog = new LazyValue<>(() -> TypeCatalog.getDefault(getExtensions()));
```

Comment:
actually... how will this work if it's no longer on the extension's catalog? How will the DSlContext get it? Did you try the app model -> xml test cases with this change?